src/main/java/com/ncedu/fooddelivery/api/v1/controllers/OrderController.java

```diff
@@ -36,46 +29,91 @@
-            @RequestParam(name = "dateEnd", required = false) @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss") Date dateEnd,
-            @RequestParam(name = "overallCost", required = false) @Digits(integer = 5, fraction = 2) @DecimalMin(value = "0.0", inclusive = false) BigDecimal overallCost,
-            @RequestParam(name = "highDemandCoeff", required = false) @Digits (integer = 1, fraction = 2) @DecimalMin("1.0") @DecimalMax("3.0") BigDecimal highDemandCoeff,
-            @RequestParam(name = "discount", required = false)  @Digits(integer = 5, fraction = 2) @DecimalMin(value = "0.0") BigDecimal discount,
-            @RequestParam(name = "promoCodeId", required = false) @Min(value = 1) @Max(value = Long.MAX_VALUE) Long promoCodeId,
-            @RequestParam(name = "clientRating", required = false) @Digits(integer = 1, fraction = 2) @DecimalMin(value = "0.0", inclusive = false) BigDecimal clientRating,
-            @RequestParam(name = "deliveryRating", required = false) @Digits (integer = 1, fraction = 2) @DecimalMin(value = "0.0", inclusive = false) BigDecimal deliveryRating,
-
+            @Valid OrderFilterDTO dto,
             @AuthenticationPrincipal User user,
             Pageable pageable){
 
-        List<OrderInfoDTO> filteredOrders;
-
-        if(user.getRole() == Role.MODERATOR){
-            Long moderatorWarehouseId = user.getModerator().getWarehouseId();
-            if(warehouseId != null){
-                if(!warehouseId.equals(moderatorWarehouseId)) throw new CustomAccessDeniedException();
-            }
-            Specification<OrderNotHierarchical> spec = OrderSpecifications.getFilterSpecification(
-                    clientId, moderatorWarehouseId, courierId, address, status, dateStart, dateEnd,
-                    overallCost, highDemandCoeff, discount, promoCodeId, clientRating,
-                    deliveryRating
-            );
-            filteredOrders = orderService.findFiltered(spec, pageable);
-
-        } else {
-            Specification<OrderNotHierarchical> spec = OrderSpecifications.getFilterSpecification(
-                    clientId, null, courierId, address, status, dateStart, dateEnd,
-                    overallCost, highDemandCoeff, discount, promoCodeId, clientRating,
-                    deliveryRating
-            );
-            filteredOrders = orderService.findFiltered(spec, pageable);
-        }
-
-        return new ResponseEntity<>(filteredOrders, HttpStatus.OK);
+        return new ResponseEntity<>(orderService.findFiltered(user, dto, pageable), HttpStatus.OK);
+    }
+
+    @PreAuthorize("hasAnyAuthority('ADMIN', 'MODERATOR')")
+    @GetMapping("/api/v1/user/{id}/orders")
+    public ResponseEntity<List<OrderInfoDTO>> getOrdersHistory(@AuthenticationPrincipal User user,
+                                                               @PathVariable @Min(value = 1) @Max(value = Long.MAX_VALUE) Long id,
```

Comment:
Хотел бы поделиться крутой штукой.  Spring может сразу искать необходимую нам сущность. Мы все также указываем при запросе на backend ID - но Spring автоматически возьмет на основе этого ID сущность которую мы укажем в PathVariable. Если он не найдет сущность - то плюнет ошибку MissingPathVariableException, которую я уже в GlobalExceptionHandler ловлю! 
```suggestion
    @GetMapping("/api/v1/user/{user}/orders")
    public ResponseEntity<List<OrderInfoDTO>> getOrdersHistory(@AuthenticationPrincipal User authedUser,
                                                               @PathVariable User user,
```